basic/src/test/java/org/jboss/hal/testsuite/test/configuration/singleton/ElectionPolicyTestCase.java

```diff
@@ -0,0 +1,206 @@
+            operations.removeIfExists(SINGLETON_POLICY_ADDRESS);
+            operations.removeIfExists(SINGLETON_POLICY_EDIT_ADDRESS);
+            SingletonSubsystemOperations.withOperations(operations).removeCacheContainer(CACHE_CONTAINER_RESOURCE_ADDRESS);
+            socketBindingResources.close();
+            administration.reloadIfRequired();
+        } finally {
+            client.close();
+        }
+    }
+
+    @Test
+    public void addSimpleElectionPolicyInUI() throws Exception {
+        page.treeNavigation()
+                .step(SINGLETON_POLICY)
+                .step(SINGLETON_POLICY_RESOURCE)
+                .step(ELECTION_POLICY)
+                .navigateToTreeItem().clickLabel();
+
+        WizardWindowWithAdvancedSelectBoxOptions window = page.getResourceManager()
+                .addResource(WizardWindowWithAdvancedSelectBoxOptions.class);
+
+        window.pick(SIMPLE)
+                .clickContinue()
+                .finish();
+
+        new ResourceVerifier(SINGLETON_POLICY_ADDRESS.and(ELECTION_POLICY, SIMPLE), client).verifyExists();
+    }
+
+    @Test
+    public void editNamePreferences() throws Exception {
+        operations.undefineAttribute(SINGLETON_POLICY_EDIT_ADDRESS.and(ELECTION_POLICY, SIMPLE), SOCKET_BINDING_PREFERENCES).assertSuccess();
+
+        page.treeNavigation()
+                .step(SINGLETON_POLICY)
+                .step(SINGLETON_POLICY_EDIT_RESOURCE)
+                .step(ELECTION_POLICY)
+                .step(SIMPLE)
+                .navigateToTreeItem().clickLabel();
+
+        new ConfigChecker.Builder(client, SINGLETON_POLICY_EDIT_ADDRESS.and(ELECTION_POLICY, SIMPLE))
+                .configFragment(page.getConfigFragment())
+                .editAndSave(ConfigChecker.InputType.TEXT, NAME_PREFERENCES, "foo\nbar\nqiz")
+                .verifyFormSaved()
+                .verifyAttribute(NAME_PREFERENCES, new ModelNode().add("foo").add("bar").add("qiz"));
+    }
+
+    @Test
+    public void editSocketBindingPreferences() throws Exception {
+        operations.undefineAttribute(SINGLETON_POLICY_EDIT_ADDRESS.and(ELECTION_POLICY, SIMPLE), NAME_PREFERENCES).assertSuccess();
+
+        page.treeNavigation()
+                .step(SINGLETON_POLICY)
+                .step(SINGLETON_POLICY_EDIT_RESOURCE)
+                .step(ELECTION_POLICY)
+                .step(SIMPLE)
+                .navigateToTreeItem().clickLabel();
+
+        new ConfigChecker.Builder(client, SINGLETON_POLICY_EDIT_ADDRESS.and(ELECTION_POLICY, SIMPLE))
+                .configFragment(page.getConfigFragment())
+                .editAndSave(ConfigChecker.InputType.TEXT, SOCKET_BINDING_PREFERENCES, socketBindingResources.getSocketBindingNamesSeparatedByNewline())
+                .verifyFormSaved()
+                .verifyAttribute(SOCKET_BINDING_PREFERENCES, socketBindingResources.getSocketBindingsAsModelNodeList());
+    }
+
+    private static final class SocketBindingResources implements AutoCloseable {
```

Comment:
Why AutoCloseable when you are not using it this way, I suggest just providing clean method, which is in this case more representative. AutoCloseable is mainly meant for resources which have allocated some resources needing to be unallocated, such as streams, connections. Not as cleanup method of what was done. You can create interface for the purpose of cleaning if you like.